jupyterlab/commands.py

```diff
@@ -1982,6 +1982,12 @@ def _compare_ranges(spec1, spec2):
     y1 = r2.set[0][0].semver
     y2 = r2.set[0][-1].semver
 
+    # Drop prereleases in comparisons to allow extension authors
+    # to not have to update their versions for each
+    # Jupyterlab prerelease version.
+    if x1.prerelease:
+        x1 = x1.inc('patch')
```

Comment:
Can we make this behavior an option in the function? As a standalone function, it's surprising that it ignores the prerelease option on just one end of one of the ranges. Would we get the same end result if we had an "increment prerelease" flag as an argument, and then just did this logic on both ends of both ranges?

Reply:
Sure, we can make this behavior optional in the standalone function.  It only has an effect on the lower end of the range, since `~2.0.0-b1` gets interpreted originally as `2.0.0-b1 <= x < 2.1.0` and converted to `2.0.0 <= x < 2.1.0`.  By doing it on the lower end of the first argument, we are saying that we don't do compatibility checks *within our own prerelease cycle only*.